### PR TITLE
YAPF conf file

### LINKED ISSUE #175: YAPF formatting issue
I'm using nvim 0.1.7, the latest vim-autoformat, and yapf 0.13.2. a minimal VIM file:
```
call plug#begin()
Plug 'Chiel92/vim-autoformat'
call plug#end()

noremap <F3> :Autoformat<CR>
```

I run `:Autoformat` on my test.py file which looks like:
```
x = {  'a':37,'b':42,

'c':927}

y = 'hello ''world'
z = 'hello '+'world'
a = 'hello {}'.format('world')
class foo  (     object  ):
  def f    (self   ):
    return       37*-+2
  def g(self, x,y=42):
      return y
def f  (   a ) :
  return      37+-+a[42-x :  y**3]
```

and get:
```
x = {'a': 37, 'b': 42,

     'c': 927}

y = 'hello ''world'
z = 'hello ' + 'world'
a = 'hello {}'.format('world')


class foo  (object):

    def f(self):
        return 37 * -+2

    def g(self, x, y=42):
        return y


def f(a):
    return 37 + -+a[42 - x:  y**3]
```

But, I expect to get (and do get when I run yapf directly on the file from the command line):
```
x = {'a': 37, 'b': 42, 'c': 927}

y = 'hello ' 'world'
z = 'hello ' + 'world'
a = 'hello {}'.format('world')


class foo(object):
    def f(self):
        return 37 * -+2

    def g(self, x, y=42):
        return y


def f(a):
    return 37 + -+a[42 - x:y**3]
```

A few other points:
* despite having a .style.yapf file in my project directory, those settings are not picked up by yapf-autoformatter (they are picked up by `yapf test.py`).
* The only message after running `:Autoformat` is: `Definition in 'g:formatdef_autopep8' was successful.`

I noticed another issue recently posted that might be related (though that user had a problem with autopep8 being run). https://github.com/Chiel92/vim-autoformat/issues/168

### PR DESCRIPTION
**Pull request by mmlb**
closes #175

### CODE REVIEW

**Comment by mmlb** [comment]
I modeled this after what clang-format does but get the following error:
```
Error detected while processing function <SNR>26_TryAllFormatters:                                                                                   
line   38:                                                                                                                                           
E110: Missing ')'                                                                                                                                    
E15: Invalid expression: g:YAPFFormatConfigFileExists() ? ('yapf -l '.a:firstline.':'.a:lastline.') : ('yapf --style="{based_on_style:'.g:formatter_yapf_style.',indent_width:'.&shiftwidth.'}" -l '.a:firstline.'-'.a:lastline)
```
I'd appreciate some help figuring it out since it looks like the parens are matched.

File: plugin/defaults.vim [hunk 1]
```diff
@@ -42,9 +42,15 @@ if !exists('g:formatter_yapf_style')
     let g:formatter_yapf_style = 'pep8'
 endif
 if !exists('g:formatdef_yapf')
-    let g:formatdef_yapf = "'yapf --style=\"{based_on_style:'.g:formatter_yapf_style.',indent_width:'.&shiftwidth.(&textwidth ? ',column_limit:'.&textwidth : '').'}\" -l '.a:firstline.'-'.a:lastline"
+    let s:configfile_def   = "'yapf -l '.a:firstline.':'.a:lastline.'"
```
Comment (Chiel92) [inline plugin/defaults.vim]: I count five single quotes. That doesn't seem right to me.

Comment (mmlb) [inline plugin/defaults.vim]: Ahhh there it is, for some reason I was concentrating on the `noconfigfile_def` line ...

**Comment by mmlb** [comment]
ping @Chiel92, I think this is good to go

File: plugin/defaults.vim [hunk 2]
```diff
@@ -42,9 +42,15 @@ if !exists('g:formatter_yapf_style')
     let g:formatter_yapf_style = 'pep8'
 endif
 if !exists('g:formatdef_yapf')
-    let g:formatdef_yapf = "'yapf --style=\"{based_on_style:'.g:formatter_yapf_style.',indent_width:'.&shiftwidth.(&textwidth ? ',column_limit:'.&textwidth : '').'}\" -l '.a:firstline.'-'.a:lastline"
+    let s:configfile_def   = "'yapf -l '.a:firstline.'-'.a:lastline"
+    let s:noconfigfile_def = "'yapf --style=\"{based_on_style:'.g:formatter_yapf_style.',indent_width:'.&shiftwidth.'}\" -l '.a:firstline.'-'.a:lastline"
```
Comment (Chiel92) [inline plugin/defaults.vim]: Why has the columnwidth feature been removed from this definition?

Comment (mmlb) [inline plugin/defaults.vim]: hmm must have been mistakenly dropped, I'll add back in

**Comment by mmlb** [comment]
fixed

**Comment by Chiel92** [comment]
Thanks!

**Comment by mmlb** [comment]
np, thanks for autoformat!

**Pull request merged**